app/services/course/assessment/session_authentication_service.rb

```diff
@@ -1,47 +1,47 @@
 # frozen_string_literal: true
-# Authenticate the assessment and store the result in the session.
+# Authenticate the assessment and update the session_id in submission.
 class Course::Assessment::SessionAuthenticationService
   # @param [Course::Assessment] assessment The password protected assessment.
   # @param [ActionDispatch::Request::Session] session The current session.
-  def initialize(assessment, session)
+  # @param [Course::Assessment::Submission|nil] submission The session id will be stored if the
+  #   submission is given.
+  def initialize(assessment, session, submission = nil)
```

Comment:
Oh okay so it's in sessions controller when there is no submission (for new submissions) 👍